### PR TITLE
Removing deprecated capture-by-reference closures

### PR DESCRIPTION
**Pull request by vpax**
This PR removes the previously deprecated form of lambda closures that captured values by reference instead of by copy.  Say goodbye to a lot of cruft!

### CODE REVIEW

**Comment by timwoj** [comment]
I'm going to hold off on merging this until 5.0 is fully released to avoid any weird merge conflicts that might arise if we need to cherry pick something over to the release branch.

File: src/Expr.cc [hunk 1]
```diff
@@ -4747,32 +4751,28 @@ LambdaExpr::LambdaExpr(std::unique_ptr<function_ingredients> arg_ing, IDPList ar
...
 	{
 	auto ft = type->AsFuncType();
 	const auto& captures = ft->GetCaptures();
 
-	capture_by_ref = false;
+	auto desc = when_parent ? "\"when\" statement" : "lambda";
 
 	if ( ! captures )
 		{
 		if ( outer_ids.size() > 0 )
```
Comment (timwoj) [inline src/Expr.cc:4763]:
```suggestion
		if ( ! outer_ids.empty() )
```

**Review by timwoj — APPROVED** [review]
One minor nit. I'll fix it during the merge.

File: src/Expr.cc (hunk 1 above)
Comment (vpax) [reply src/Expr.cc:4763]: Oops yeah, thx.  I try to canonicalize those but sometimes miss some!

**Pull request merged**